cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -442,6 +488,31 @@ private predicate simpleInstructionLocalFlowStep(Instruction iFrom, Instruction
   or
   iTo.(PhiInstruction).getAnOperand().getDef() = iFrom
   or
+  // A read side effect is almost never exact since we don't know exactly how
+  // much memory the callee will read.
+  iTo.(ReadSideEffectInstruction).getSideEffectOperand().getAnyDef() = iFrom and
+  not iFrom.isResultConflated()
```

Comment:
I don't like this rule. It sends flow into a `ReadSideEffectInstruction`, which actually means there is flow to the result value of that instruction. But it's an instruction without a result!

The rule is here because we need a node to be the `ArgumentNode` in `DataFlowPrivate.qll`, and I've chosen the node for the `ReadSideEffectInstruction`. The alternatives I can see are:
1. Omit this rule and use the most recent definition of the indirection (here named `iFrom`) as the `ArgumentNode`. Then a node could be the argument for multiple calls, leading to confusing path explanations.
1. Create a new synthetic node for this purpose.
1. Change the data-flow library so that flow alternates between `Instruction` and `Operand` nodes. The list of good reasons to do this is starting to get long.

Reply:
You can see the effect on path explanations in `argvLocal.expected`, where many new nodes with the awkward name `BufferReadSideEffect` are now there.